misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/JBallerinaDebugServer.java

```diff
@@ -282,6 +283,10 @@ ClientConfigHolder getClientConfigHolder() {
             if (loadedThreadFrames.containsKey(activeThread.uniqueID())) {
                 stackTraceResponse.setStackFrames(loadedThreadFrames.get(activeThread.uniqueID()));
             } else {
+                if (activeThread.frames().get(0).location().method().name().equals(METHOD_INIT)) {
+                    prepareFor(DebugInstruction.STEP_OVER);
+                    eventProcessor.sendStepRequest(args.getThreadId(), StepRequest.STEP_OVER);
+                }
```

Comment:
I suppose we should do a `step-in` request in here. Am I missing something?

Reply:
Also better to add a comment explaining **why** we are doing this, for the future reference.

Reply:
Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/31506/commits/e88b66976d90596c9a5915accfdad27204e71e8b